exporter/elasticsearchexporter/exporter.go

```diff
@@ -7,61 +7,83 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"runtime"
 	"time"
 
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/ptrace"
-	"go.uber.org/zap"
 )
 
 type elasticsearchExporter struct {
-	logger *zap.Logger
+	component.TelemetrySettings
+	userAgent string
 
+	config         *Config
 	index          string
 	logstashFormat LogstashFormatSettings
 	dynamicIndex   bool
+	model          mappingModel
 
-	client      *esClientCurrent
 	bulkIndexer *esBulkIndexerCurrent
-	model       mappingModel
 }
 
-func newExporter(logger *zap.Logger, cfg *Config, index string, dynamicIndex bool) (*elasticsearchExporter, error) {
+func newExporter(
+	cfg *Config,
+	set exporter.CreateSettings,
+	index string,
+	dynamicIndex bool,
+) (*elasticsearchExporter, error) {
 	if err := cfg.Validate(); err != nil {
 		return nil, err
 	}
 
-	client, err := newElasticsearchClient(logger, cfg)
-	if err != nil {
-		return nil, err
-	}
-
-	bulkIndexer, err := newBulkIndexer(logger, client, cfg)
-	if err != nil {
-		return nil, err
-	}
-
 	model := &encodeModel{
 		dedup: cfg.Mapping.Dedup,
 		dedot: cfg.Mapping.Dedot,
 		mode:  cfg.MappingMode(),
 	}
 
+	userAgent := fmt.Sprintf(
+		"%s/%s (%s/%s)",
+		set.BuildInfo.Description,
+		set.BuildInfo.Version,
+		runtime.GOOS,
+		runtime.GOARCH,
+	)
+
 	return &elasticsearchExporter{
-		logger:      logger,
-		client:      client,
-		bulkIndexer: bulkIndexer,
+		TelemetrySettings: set.TelemetrySettings,
+		userAgent:         userAgent,
 
+		config:         cfg,
 		index:          index,
 		dynamicIndex:   dynamicIndex,
 		model:          model,
 		logstashFormat: cfg.LogstashFormat,
 	}, nil
 }
 
+func (e *elasticsearchExporter) Start(ctx context.Context, host component.Host) error {
+	client, err := newElasticsearchClient(ctx, e.config, host, e.TelemetrySettings, e.userAgent)
```

Comment:
👍  double-checked and currently the ES client is implicitly closed so no need to defend here.